services/core-api/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -178,6 +179,10 @@ def post(self, mine_guid):
                     rec_string, mine_incident_id=incident.mine_incident_id)
                 new_recommendation.save()
 
+        mine_incident_category_codes = data.get('mine_incident_category_xref')
+        MineIncidentCategoryXref.create_batch(incident.mine_incident_id,
+                                              mine_incident_category_codes)
+
```

Comment:
Code to create the incident types in the incident record (POST).

Reply:
this should almost identically mimic how `dangerous_occurrence_subparagraph_ids` which has identical behaviour and shouldn't have many edge cases

Reply:
Ok, I'll see how it compares to that. I was using 'recommendations' stuff as a reference a lot. I believe it is also structured similarily?